libnavigation-ui/src/test/java/com/mapbox/navigation/ui/camera/NavigationCameraTest.java

```diff
@@ -184,6 +191,62 @@ public void update_overrideSetsLocationComponentCameraModeNone() {
     verify(locationComponent).setCameraMode(eq(CameraMode.NONE));
   }
 
+  @Test
+  public void onShowRouteOverview_routeProgressAndRouteIsNull() {
+    MapboxMap mapboxMap = mock(MapboxMap.class);
+    MapboxNavigation navigation = mock(MapboxNavigation.class);
+    NavigationCamera camera = buildCamera(mapboxMap, navigation);
+
+    camera.start(null);
+    camera.showRouteOverview(new int[4]);
+
+    // verify no crash
+  }
+
+  @Test
+  public void onShowRouteOverview_routeProgressIsNull() {
+    MapboxMap mapboxMap = mock(MapboxMap.class);
+    MapboxNavigation navigation = mock(MapboxNavigation.class);
+    DirectionsRoute directionsRoute = mock(DirectionsRoute.class);
+    DynamicCamera dynamicCamera = mock(DynamicCamera.class);
+    NavigationCamera camera = buildCamera(mapboxMap, navigation);
+    camera.setCamera(dynamicCamera);
+    List<Point> points = new ArrayList<>();
+    points.add(mock(Point.class));
+    points.add(mock(Point.class));
+    points.add(mock(Point.class));
+    when(dynamicCamera.overview(any())).thenReturn(points);
+
+    camera.start(directionsRoute);
+    camera.showRouteOverview(new int[4]);
+
+    verify(mapboxMap).animateCamera(any(), anyInt(), any());
+  }
+
+  @Test
+  public void onShowRouteOverview_routeInfoIsNull() {
```

Comment:
NIT

```suggestion
  public void onShowRouteOverview_whenRouteInfoIsNull() {
```

Maybe clearer? 🤔 

```suggestion
  public void animateCameraIsCalledWhenShowRouteOverviewIfRouteInfoIsNull() {
```